ops/charm.py

```diff
@@ -261,62 +344,157 @@ class RelationCreatedEvent(RelationEvent):
 
 
 class RelationJoinedEvent(RelationEvent):
-    """Represents the `relation-joined` hook from Juju.
-
-    This is triggered whenever a new unit of a related application joins the relation.
-    (eg, a unit was added to an existing related app, or a new relation was established
-    with an application that already had units.)
+    """Event triggered when a new unit joins a relation.
+
+    This event is triggered whenever a new unit of a related
+    application joins the relation.  The event fires only when that
+    remote unit is first observed by the unit. Callback methods bound
+    to this event may set any local unit settings that can be
+    determined using no more than the name of the joining unit and the
+    remote `private-address` setting, which is always available when
+    the relation is created and is by convention not deleted.
     """
 
 
 class RelationChangedEvent(RelationEvent):
-    """Represents the `relation-changed` hook from Juju.
-
-    This is triggered whenever there is a change to the data bucket for a related
-    application or unit. Look at `event.relation.data[event.unit/app]` to see the
-    new information.
+    """Event triggered when relation data changes.
+
+    This event is triggered whenever there is a change to the data
+    bucket for a related application or unit. Look at
+    `event.relation.data[event.unit/app]` to see the new information,
+    where `event` is the event object passed to the callback method
+    bound to this event.
+
+    This event always fires once, after :class: `RelationJoinedEvent`
+    event, and will subsequently be fire whenever that remote unit
+    changes its settings for the relation. Callback methods bound to
+    this event should be the only ones that rely on remote relation
+    settings. They should not error if the settings are incomplete,
+    since it can be guaranteed that when the remote unit changes its
+    settings, the event will be fire again.
```

Comment:
It fires if either the unit data or the application data for the relation have changed.

Reply:
Thanks changed to "remote unit or application changes".